api/trace/trace_context_propagator.go

```diff
@@ -57,7 +57,11 @@ func (TraceContext) Inject(ctx context.Context, supplier propagation.HTTPSupplie
 }
 
 func (tc TraceContext) Extract(ctx context.Context, supplier propagation.HTTPSupplier) context.Context {
-	return ContextWithRemoteSpanContext(ctx, tc.extract(supplier))
+	sc := tc.extract(supplier)
```

Comment:
If instead we had `TraceContext.extract` return a bool value as a second return value, we could avoid the byte array comparison in `TraceID.IsValid`. Did you consider that alternative?

Reply:
Not really. The bool value would provide a redundant information, together with an invalid span context. And my thinking is that checking `TraceID.IsValid` is not much of a bottleneck - it's likely that golang is basically using some optimized-to-death version of memcmp. At least I haven't profiled it yet. :)

Reply:
Fair enough. I appreciate the discussion.